### PR TITLE
Added ORAPI DB to Terraform Config

### PR DESCRIPTION
**Pull request by spartanbear**
Added ORAPI DB to Terraform Config for all environments. Prod has been commented out

### CODE REVIEW

File: Azure/Dev/resource_sql.tf [hunk 1]
```diff
@@ -36,6 +40,7 @@ resource "azurerm_sql_firewall_rule" "bc-sql-pri" {
   end_ip_address      = "0.0.0.0"
 }
 
+#SQL Database using for the BuyingCatalgueService Private
```
Comment (stuart-bass-cgi) [inline Azure/Dev/resource_sql.tf]: typo BuyingCatalogueService, but this is dev, so no public / private

File: Azure/Dev/resource_sql.tf [hunk 2]
```diff
@@ -44,9 +49,9 @@ resource "azurerm_sql_database" "sql-bapi-pri" {
   collation                        = var.sql_collation
   edition                          = var.sql_edition
   requested_service_objective_name = var.sql_size
-
 }
 
+#SQL Database using for the BuyingCatalgueIdentityService
```
Comment (stuart-bass-cgi) [inline Azure/Dev/resource_sql.tf]: Catalogue

File: Azure/Dev/resource_sql.tf [hunk 3]
```diff
@@ -55,14 +60,26 @@ resource "azurerm_sql_database" "sql-isapi" {
   collation                        = var.sql_collation
   edition                          = var.sql_edition
   requested_service_objective_name = var.sql_size
+}
 
+#SQL Database using for the BuyingCatalgueOrderingService
```
Comment (stuart-bass-cgi) [inline Azure/Dev/resource_sql.tf]: Catalogue

File: Azure/Dev/resource_sql.tf [hunk 4]
```diff
@@ -55,14 +60,26 @@ resource "azurerm_sql_database" "sql-isapi" {
...
+  resource_group_name              = azurerm_resource_group.bc-sql-pri.name
+  location                         = var.region
+  server_name                      = azurerm_sql_server.bc-sql-pri.name
+  collation                        = var.sql_collation
+  edition                          = var.sql_edition
+  requested_service_objective_name = var.sql_size
 }
 
+#Storage Account used to store the SQL activity logs
 resource "azurerm_advanced_threat_protection" "bc-sql-pri" {
```
Comment (stuart-bass-cgi) [inline Azure/Dev/resource_sql.tf:68]: This isn't the storage account, but threat protection for it.

File: Azure/Dev/resource_sql.tf [hunk 5]
```diff
@@ -6,6 +7,7 @@ resource "azurerm_resource_group" "bc-sql-pri" {
   }
 }
 
+#Secondary failover Azure Resource Group storing both the SQL Server & DB within UKWest
 resource "azurerm_resource_group" "bc-sql-sec" {
```
Comment (stuart-bass-cgi) [inline Azure/Dev/resource_sql.tf]: do we have secondary failover for dev?

File: Azure/Prod/resource_sql.tf [hunk 6]
```diff
@@ -48,6 +54,7 @@ resource "azurerm_sql_firewall_rule" "bc-sql-sec" {
   end_ip_address      = "0.0.0.0"
 }
 
+#SQL Database using for the BuyingCatalgueService Private
```
Comment (stuart-bass-cgi) [inline Azure/Prod/resource_sql.tf]: Catalogue (not going to repeat this comment - looks like a copy/paste error everywhere

File: Azure/Prod/resource_sql.tf [hunk 7]
```diff
@@ -67,19 +75,43 @@ resource "azurerm_sql_database" "sql-bapi-pub" {
   collation                        = var.sql_collation
   edition                          = var.sql_edition
   requested_service_objective_name = var.sql_size
-
 }
```
Comment (stuart-bass-cgi) [inline Azure/Prod/resource_sql.tf:79]: Probably shouldn't have commented out code

File: Azure/Prod/resource_sql.tf [hunk 8]
```diff
@@ -67,19 +75,43 @@ resource "azurerm_sql_database" "sql-bapi-pub" {
...
+#   resource_group_name              = azurerm_resource_group.bc-sql-pri.name
+#   location                         = var.region
+#   server_name                      = azurerm_sql_server.bc-sql-pri.name
+#   collation                        = var.sql_collation
+#   edition                          = var.sql_edition
+#   requested_service_objective_name = var.sql_size
+# }
+
+#Storage Account used to store the SQL activity logs
 resource "azurerm_advanced_threat_protection" "bc-sql-pri" {
```
Comment (stuart-bass-cgi) [inline Azure/Prod/resource_sql.tf:103]: threat protection settings for the storage account

File: Azure/Prod/resource_sql.tf [hunk 9]
```diff
@@ -67,19 +75,43 @@ resource "azurerm_sql_database" "sql-bapi-pub" {
...
   target_resource_id = azurerm_storage_account.sqluks.id
   enabled            = true
 }
 
+#Storage Account used to store the SQL activity logs
 resource "azurerm_advanced_threat_protection" "bc-sql-sec" {
```
Comment (stuart-bass-cgi) [inline Azure/Prod/resource_sql.tf:109]: threat protection settings for the storage account

File: Azure/Prod/resource_sql.tf [hunk 10]
```diff
@@ -94,8 +126,9 @@ resource "azurerm_sql_failover_group" "sql-bapi-pri" {
   }
 }
 
+#Failover config for BuyingCatalgueService Private
 resource "azurerm_sql_failover_group" "sql-bapi-pub" {
-  name                = "${var.project}-${var.environment}-sql-fog1"
+  name                = "${var.project}-${var.environment}-sql-fog"
```
Comment (stuart-bass-cgi) [inline Azure/Prod/resource_sql.tf]: won't this name conflict?

File: Azure/Prod/resource_sql.tf [hunk 11]
```diff
@@ -108,6 +141,37 @@ resource "azurerm_sql_failover_group" "sql-bapi-pub" {
   }
 }
```
Comment (stuart-bass-cgi) [inline Azure/Prod/resource_sql.tf:143]: commented out code

File: Azure/Test/resource_sql.tf [hunk 12]
```diff
@@ -80,16 +89,31 @@ resource "azurerm_sql_database" "sql-isapi" {
...
+}
+
+#Storage Account used to store the SQL activity logs
 resource "azurerm_advanced_threat_protection" "bc-sql-pri" {
   target_resource_id = azurerm_storage_account.sqluks.id
   enabled            = true
 }
 
+#Storage Account used to store the SQL activity logs
 resource "azurerm_advanced_threat_protection" "bc-sql-sec" {
```
Comment (stuart-bass-cgi) [inline Azure/Test/resource_sql.tf:111]: threat protection settings for the storage account

File: Azure/Test/resource_sql.tf [hunk 13]
```diff
@@ -80,16 +89,31 @@ resource "azurerm_sql_database" "sql-isapi" {
...
+  resource_group_name              = azurerm_resource_group.bc-sql-pri.name
+  location                         = var.region
+  server_name                      = azurerm_sql_server.bc-sql-pri.name
+  collation                        = var.sql_collation
+  edition                          = var.sql_edition
+  requested_service_objective_name = var.sql_size
+
+}
+
+#Storage Account used to store the SQL activity logs
```
Comment (stuart-bass-cgi) [inline Azure/Test/resource_sql.tf]: threat protection settings for the storage account

**Review by stuart-bass-bjss — CHANGES_REQUESTED** [review]

File: Azure/Prod/resource_sql.tf (hunk 7 above)
Comment (stuart-bass-cgi) [reply Azure/Prod/resource_sql.tf:79]: Going to live with it for now.

**Review by stuart-bass-bjss — APPROVED** [review]

**Pull request merged**